### DISCUSSION TITLE
请问一下，可以实现禁止打开工作区部分文件吗？

### DISCUSSION THREAD

**Discussion by tsuyoii** [opensumi/core#2253] [Q&A]
请问一下，可以实现禁止打开工作区部分文件吗？
具体就是：工作区目录中展示的文件只有一部分是可以打开并编辑的，另一部分文件是灰色，不可点击(不可进入查看或编辑该文件)

**Comment by erha19** [comment]
@tsuyoii 如果需要完全实现你提到的实现，可能需要修改 `@opensumi/file-tree-next` 模块部分实现，但如果简单实现，你可以将这些不可被打开的文件通过 `files.exclude` 配置，在文件树中屏蔽一类文件的展示，如在打开工作区前通过在 AppConfig 方式设置项目默认配置，如：

```ts
appconfig = {
   ...
  defaultPreferences: {
    'files.exclude': {
         // 默认值 glob 表达式
         '**/.git': true,
         '**/.svn': true,
         '**/.hg': true,
         '**/CVS': true,
         '**/.DS_Store': true,
        // 追加, 这里屏蔽了 index.js 文件的展示
        '**/index.js'
    }
  }
   ...
}
```


同时，如果期望实现部分文件的只读，也可以参考 [自定义只读模式
](https://opensumi.com/zh/docs/integrate/universal-integrate-case/custom-readonly-mode) 文档

**Reply by tsuyoii** [reply]
感谢回复，我之前也考虑过直接隐藏不显示，但需求方还是希望展示出来，谢谢回复，我去查看一下@opensumi/file-tree-next模块
顺便问一下，您们会考虑以后添加这个功能吗

**Reply by erha19** [reply]
这块目前没有计划实现

节点渲染的逻辑可以看这里 
https://github.com/opensumi/core/blob/3340fa89eab63413a00335f52a81d6cfa20dd2e5/packages/file-tree-next/src/browser/file-tree-node.tsx#L61

处理节点点击的逻辑
https://github.com/opensumi/core/blob/4854f4fe3a3117dcbfca0292497c9a69849c8e3c/packages/file-tree-next/src/browser/services/file-tree-model.service.ts#L808

**Reply by tsuyoii** [reply]
好的，谢谢

**Comment by hacke2** [comment]
fileService 里提供了文件读取黑名单的参数配置，可以禁止文件系统对此类文件的读取，可以参考下

```js

new ServerApp({
    ...,
    blockPatterns: [
        // 禁止在 IDE 里访问到 /release 下的代码
        'file:///release/{app,config,run}/**',
      ],
})

```

**Reply by tsuyoii** [reply]
好呢，谢谢
